framework_crates/bones_framework/src/networking/socket.rs

```diff
@@ -0,0 +1,300 @@
+                                            ),
+                                        )
+                                        .await;
+                                    }
+                                    if ggrs_sender.send((i, message)).await.is_err() {
+                                        break;
+                                    }
+                                }
+                                Err(e) => {
+                                    warn!("Connection error: {e}");
+                                }
+                            }
+                        }
+                    }
+                });
+
+                // Reliable message receiver
+                let reliable_sender = reliable_sender.clone();
+                RUNTIME.spawn(async move {
+                    #[cfg(feature = "debug-network-slowdown")]
+                    use turborand::prelude::*;
+                    #[cfg(feature = "debug-network-slowdown")]
+                    let rng = AtomicRng::new();
+
+                    loop {
+                        tokio::select! {
+                            closed = conn.closed() => {
+                                warn!("Connection error: {closed}");
+                                break;
+                            }
+                            result = conn.accept_uni() => match result {
+                                Ok(mut stream) => {
+                                    let data = stream.read_to_end(4096).await.expect("Network read error");
+
+                                    // Debugging code to introduce artificial latency
+                                    #[cfg(feature = "debug-network-slowdown")]
+                                    {
+                                        use async_timer::Oneshot;
+                                        async_timer::oneshot::Timer::new(
+                                            std::time::Duration::from_millis(
+                                                (rng.f32_normalized() * 100.0) as u64 + 1,
+                                            ),
+                                        )
+                                        .await;
+                                    }
+                                    if reliable_sender.send((i, data)).await.is_err() {
+                                        break;
+                                    }
+                                }
+                                Err(e) => {
+                                    warn!("Connection error: {e}");
+                                }
+                            },
+                        }
+                    }
+                });
+            }
+        }
+
+        Self {
+            player_idx,
+            player_count: connections.iter().flatten().count() + 1,
+            connections,
+            ggrs_receiver,
+            reliable_receiver,
+            match_id: 0,
+        }
+    }
+}
+
+impl NetworkSocket for Socket {
```

Comment:
`Socket` is the only implementor of this trait now, could be removed as well, but that's your call